detectors/node/opentelemetry-resource-detector-docker/test/DockerCGroupv1Detector.test.ts

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as sinon from 'sinon';
+import * as assert from 'assert';
+import { Resource } from '@opentelemetry/resources';
+import { dockerCGroupV1Detector } from '../src';
+import {
+  assertContainerResource,
+  assertEmptyResource,
+} from '@opentelemetry/contrib-test-utils';
+
+import { DockerCGroupV1Detector } from '../src';
+
+describe('dockerCGroupV1Detector', () => {
+  let readStub;
+  const correctCgroupData =
+    'abcdefghijklmnopqrstuvwxyzabcdefghijklmnopqrstuvwxyzabcdefghijklm';
+
+  afterEach(() => {
+    sinon.restore();
+  });
+
+  describe('Supported docker - Container ID ', () => {
+    it('should return a emoty resource - docker cgroup v1 detector', async () => {
+      const resource: Resource = await dockerCGroupV1Detector.detect();
```

Comment:
Sorry - I don't understand this comment - can you explain a little more
It should return { attributes: {} } as resource if we do not give it the right file path ( in case of test - the right stub which resolves into the container id  ) ?

Reply:
Maybe it's not really a problem 🤷‍♂️, but I can't understand why the invocation of this detector does not return non-empty resource when run in CI.

Reply:
I have added a stub to return undefined - just to be more careful

Reply:
Does the resource detector require some sort of elevated permission or security context? I'm also curious why this returns undefined in CI because our CI runs inside docker containers.

Reply:
what version of docker is it running on ? if it is v2 - then this is expected to return undefined as the cgroup path is different and may need permission.